src/aspire/classification/class_selection.py

```diff
@@ -107,11 +138,507 @@ def __init__(self, seed=None):
         """
         self.seed = seed
 
-    def _select(self, n, classes, reflections, distances):
+    def _select(self, classes, reflections, distances):
         """
         Select random `n` classes from the population.
         """
+        # Assign uniform quality.
+        self._quality_scores = np.zeros(self.n)
+
         # Instantiate a random Generator
         rng = np.random.default_rng(self.seed)
         # Generate and return indices for random sample
-        return rng.choice(np.max(classes[:, 0]), size=n, replace=False)
+        # +1 for zero indexing
+        return rng.choice(self.n, size=self.n, replace=False)
+
+
+class ContrastClassSelector(ClassSelector):
+    """
+    Selects top classes based on highest contrast,
+    as estimated by variances of `distances`.
+
+    Note that `distances` is the Nearest Neighbors distances,
+    and in the case of RIR this is a small rotationally invariant feature
+    vector.  For methods based on class average images, see subclasses of `GlobalClassSelector`.
+    """
+
+    def _select(self, classes, reflections, distances):
+        # Compute per class variance
+        dist_var = np.var(distances, axis=1)
+
+        # Compute the ordering, descending
+        sorted_class_inds = np.argsort(dist_var)[::-1]
+        # Store the sorted quality scores (maps to selection output).
+        self._quality_scores = dist_var[sorted_class_inds]
+
+        # Return indices
+        return sorted_class_inds
+
+
+class DistanceClassSelector(ClassSelector):
+    """
+    Selects top classes based on lowest mean distance
+    as estimated by `distances`.
+
+    Note that `distances` is the Nearest Neighbors distances,
+    and in the case of RIR this is a small rotationally invariant feature
+    vector.  For methods based on class average images, see subclasses of `GlobalClassSelector`.
+    """
+
+    def _select(self, classes, reflections, distances):
+        # Compute per class variance.
+        # Skips self distance (0).
+        dist_mean = np.mean(distances[:, 1:], axis=1)
+
+        # Compute the ordering, descending
+        sorted_class_inds = np.argsort(dist_mean)[::-1]
```

Comment:
This might have been from when I was using the `dot` from legacy code.. but I think I remember the worst class average's neighbors all looking really close to each other...  because just noise...  which is probably opposite of your intuition.  Anyway will double check. The network `distances` are more consistent now.

Reply:
This was from when I was using the legacy `dot`! Changed, thanks!